### PR TITLE
[BugFix] Fix sync issue

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
https://github.com/StarRocks/starrocks/issues/12324
## Problem Summary(Required) ：

This PR fixes two issues:
1. After the segment file is written, add a sync operation to ensure the durability of data.
2. Sync tablet's parent directories after creating a tablet.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/fs/fs_posix.cpp [hunk 1]
```diff
@@ -214,7 +214,6 @@ class PosixWritableFile : public WritableFile {
             RETRY_ON_EINTR(ret, ftruncate(_fd, _filesize));
             if (ret != 0) {
                 s = io_error(_filename, errno);
-                _pending_sync = true;
```
Comment (trueeyu) [inline be/src/fs/fs_posix.cpp:217]: Why remove the _pending_sync?

File: be/src/storage/data_dir.cpp [hunk 2]
```diff
@@ -156,7 +156,20 @@ Status DataDir::get_shard(uint64_t* shard) {
     }
     shard_path_stream << _path << DATA_PREFIX << "/" << next_shard;
     std::string shard_path = shard_path_stream.str();
+    // First check whether the shard path exists. If it does not exist, sync the data directory.
+    bool sync_data_path = false;
+    if (!fs::path_exist(shard_path)) {
+        sync_data_path = true;
+    }
     RETURN_IF_ERROR(_fs->create_dir_recursive(shard_path));
+    if (sync_data_path) {
```
Comment (trueeyu) [inline be/src/storage/data_dir.cpp:165]: Why add `sync_data_path`?

**Review by decster — COMMENTED** [review]
should also sync dir every time a new segment file is created.

File: be/src/fs/fs_posix.cpp [hunk 3]
```diff
@@ -258,10 +257,7 @@ class PosixWritableFile : public WritableFile {
     }
 
     Status sync() override {
-        if (_pending_sync) {
-            _pending_sync = false;
```
Comment (chaoyli) [inline be/src/fs/fs_posix.cpp:262]: _pending_sync can guarantee only sync upon close()

**Review by chaoyli — APPROVED** [review]

**Review by decster — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by decster** [comment]
@mergify backport branch-2.4

**Pull request merged**